index.js

```diff
@@ -136,35 +175,48 @@ function nowServing(line) {
 }
 
 const line = []
-
+let i = 0
 // ***** Scope & Closures - Question 1 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+function takeATicketNumber(array){
+  i = i + 1
+  line.push(i)
+  
+  return `Welcome, you are ticket number ${i}`
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+function nowServing(array){
+  return `Currently serving ${line.shift()}`
+}
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
 
 // const newLine = []
+
+console.log("IDK what this question is even asking lol")
```

Comment:
lol
This function wants you to return a function that uses closure, so the solution would look something like this:
```
function ticketNumberGeneratorFunc(){
   // ...
   return function(){
   }
}
```